### PR TITLE
Various refactoring / enhancements for the upcoming centos7 formula

### PR DESCRIPTION
**Pull request by philpep**

### CODE REVIEW

File: run-tests.py [hunk 1]
```diff
@@ -68,14 +68,19 @@ def dev(args, remain):
...
+    cmd = [
+        "docker", "run", "-d",
         "-v", "{0}/test/minion.conf:/etc/salt/minion.d/minion.conf".format(BASEDIR),
         "-v", "{0}/test/salt:/srv/salt".format(BASEDIR),
         "-v", "{0}/test/pillar:/srv/pillar".format(BASEDIR),
         "-v", "{0}/{1}:/srv/formula/{1}".format(BASEDIR, _formula),
-        tag, "/bin/bash",
-    ])
+    ]
+    cmd.append(tag)
```
Comment (dlax) [inline run-tests.py]: previous version (no `.append`) looks nicer.

File: run-tests.py [hunk 2]
```diff
@@ -68,14 +68,19 @@ def dev(args, remain):
...
+    docker_id = subprocess.check_output(cmd).strip()
+    try:
+        subprocess.call(["docker", "exec", "-it", docker_id, "/bin/bash"])
+    finally:
+        subprocess.call(["docker", "rm", "-f", docker_id])
```
Comment (dlax) [inline run-tests.py]: I don't get the point of all this, nor how this is related to what's mentioned in the commit message. Perhaps add a comment to explain?

File: test/test_saemref.py [hunk 3]
```diff
@@ -34,3 +63,22 @@ def test_idempotence(Salt, state, exclude):
...
+
+    cubicweb = Process.get(ppid=supervisord.pid)
+    assert cubicweb.comm == "cubicweb-ctl"
+    assert cubicweb.user == "saemref"
+    assert cubicweb.group == "saemref"
+
+    assert Socket("tcp://0.0.0.0:8080").is_listening
+
+    html = Command.check_output("curl http://localhost:8080")
+    assert "<title>accueil (Référentiel SAEM)</title>" in html
```
Comment (dlax) [inline test/test_saemref.py]: Unrelated with commit message, it seems.

**Comment by dlax** [comment]
Split the commit I commented on and perhaps add a bit of doc about mysterious docker commands in `run-tests.py`.

File: run-tests.py (hunk 1 above)
Comment (philpep) [inline run-tests.py]: Fixed

File: run-tests.py (hunk 2 above)
Comment (philpep) [inline run-tests.py]: Added comments

File: test/test_saemref.py (hunk 3 above)
Comment (philpep) [inline test/test_saemref.py]: Added an explanation in the commits messsage (2ddf7ab75a4d07a8036c4a5e4951ad8481fa2f0c)

File: run-tests.py [hunk 4]
```diff
@@ -68,14 +68,21 @@ def dev(args, remain):
...
         "-v", "{0}/{1}:/srv/formula/{1}".format(BASEDIR, _formula),
-        tag, "/bin/bash",
-    ])
+        tag,
+    ]
+    # Run the container default CMD as pid 1 (init system)
+    docker_id = subprocess.check_output(cmd).strip()
+    try:
+        # Spawn a interactive shell in the container
+        subprocess.call(["docker", "exec", "-it", docker_id, "/bin/bash"])
```
Comment (dlax) [inline run-tests.py]: why do you do this? (I understand that `docker exec -it ... /bin/bash` command spawns a shell, the point is to explain why this is useful ;)

File: test/test_saemref.py (hunk 3 above)
Comment (dlax) [inline test/test_saemref.py]: Philippe Pepiot a écrit :
> In test/test_saemref.py
> <https://github.com/logilab/saemref-formula/pull/2#discussion_r62469024>:
>
>> +def test_saemref_running(Process, Service, Socket, Command):
>> +    assert Service("supervisord").is_enabled
>> +
>> +    supervisord = Process.get(comm="supervisord")
>> +    assert supervisord.user == "saemref"
>> +    assert supervisord.group == "saemref"
>> +
>> +    cubicweb = Process.get(ppid=supervisord.pid)
>> +    assert cubicweb.comm == "cubicweb-ctl"
>> +    assert cubicweb.user == "saemref"
>> +    assert cubicweb.group == "saemref"
>> +
>> +    assert Socket("tcp://0.0.0.0:8080").is_listening
>> +
>> +    html = Command.check_output("curl http://localhost:8080")
>> +    assert "<title>accueil (Référentiel SAEM)</title>" in html
>
> Added an explanation in the commits messsage (2ddf7ab
> <https://github.com/logilab/saemref-formula/commit/2ddf7ab75a4d07a8036c4a5e4951ad8481fa2f0c>)
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/logilab/saemref-formula/pull/2/files/135229ef1dea7e9cbfb4b6022237203eab7b8ce9..47315c3cc34cea968455fb6bb5e155b812bea117#r62469024>
>

Ok

File: run-tests.py (hunk 4 above)
Comment (philpep) [inline run-tests.py]: The above command run ``/sbin/init`` doesn't give a shell (the purpose of the ``./run-tests.py dev`` command).

I splitted the concerned commit into 8621614, 253447d and 253447d hope it make things more clear and relevant to you ;)

Comment (dlax) [inline run-tests.py]: Philippe Pepiot a écrit :
> The above command run |/sbin/init| doesn't give a shell (the purpose of
> the |./run-tests.py dev| command).
>

Ok, I remember now this "dev" command. A docstring would have been nice 
(or use click?).
